src/tract.h

```diff
@@ -1,203 +1,203 @@
+} ITEM;                         /* (item) */
+
+typedef struct {                /* --- a transaction --- */
+  int     cnt;                  /* number of items */
+  int     items[1];             /* item identifier vector */
+} TRACT;                        /* (transaction) */
+
+typedef struct {                /* --- an itemset --- */
+  TFSCAN  *tfscan;              /* table file scanner */
+  char    chars[4];             /* special characters */
+  NIMAP   *nimap;               /* name/identifier map */
+  int     app;                  /* default appearance indicator */
+  int     vsz;                  /* size of transaction buffer */
+  int     cnt;                  /* number of items in transaction */
+  int     *items;               /* items in transaction */
+} ITEMSET;                      /* (item set) */
+
+typedef struct {                /* --- a transaction set --- */
+  ITEMSET *itemset;             /* underlying item set */
+  int     max;                  /* maximum number of items per t.a. */
+  int     vsz;                  /* size of transaction vector */
+  int     cnt;                  /* number of transactions */
+  int     total;                /* total number of items */
+  TRACT   **tracts;             /* transaction vector */
+} TASET;                        /* (transaction set) */
+
+typedef struct _tatree {        /* --- a transaction tree (node) --- */
+  int     cnt;                  /* number of transactions */
+  int     max;                  /* size of largest transaction */
+  int     size;                 /* node size (number of children) */
+  int     items[1];             /* next items in rep. transactions */
+} TATREE;                       /* (transaction tree) */
+
+/*----------------------------------------------------------------------
+  Item Set Functions
+----------------------------------------------------------------------*/
+extern ITEMSET*    is_create  (void);
+extern void        is_delete  (ITEMSET *iset);
+extern TFSCAN*     is_tfscan  (ITEMSET *iset);
+extern void        is_chars   (ITEMSET *iset, const char *blanks,
+                                              const char *fldseps,
+                                              const char *recseps,
+                                              const char *cominds);
+
+extern int         is_cnt     (ITEMSET *iset);
+extern int         is_item    (ITEMSET *iset, const char *name);
+extern const char* is_name    (ITEMSET *iset, int item);
+extern int         is_getfrq  (ITEMSET *iset, int item);
+extern int         is_setfrq  (ITEMSET *iset, int item, int frq);
+extern int         is_addfrq  (ITEMSET *iset, int item, int frq);
+extern int         is_getapp  (ITEMSET *iset, int item);
+extern int         is_setapp  (ITEMSET *iset, int item, int app);
+
+extern int         is_readapp (ITEMSET *iset, FILE *file);
+extern int         is_read    (ITEMSET *iset, FILE *file);
+
+extern int         is_recode  (ITEMSET *iset, int minfrq, int dir,

```

Comment:
/* Makh2018: Adding "mode" and "fullS" as parameters to remove full support items in case of generator itemsets mining*/